File: FWPopupView_OC/FWPopupView_OC/FWDemoViewController.m
//
//  FWDemoViewController.m
//  FWPopupView
//
//  Created by xfg on 2018/3/27.
//  Copyright © 2018年 xfg. All rights reserved.
//

#import "FWDemoViewController.h"
#import <FWPopupView/FWPopupView-Swift.h>
#import "FWCustomDemoVC.h"
#import "FWMenuViewDemoVC.h"

@interface FWDemoViewController ()

@property (nonatomic, strong) NSArray       *titleArray;
@property (nonatomic, strong) FWAlertView   *alertWithImageView;
@property (nonatomic, strong) FWCustomSheetView *customSheetView;
@property (nonatomic, strong) FWCustomSheetView *customSheetView2;

@end

@implementation FWDemoViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.navigationItem.title = @"FWPopupView";
    
    self.titleArray = @[@"Alert - 单个按钮", @"Alert - 两个按钮", @"Alert - 两个按钮（修改参数）", @"Alert - 多个按钮", @"Alert - 带输入框", @"Alert - 带自定义视图", @"Sheet - 少量Item", @"Sheet - 大量Item", @"Date - 自定义日期选择", @"Menu - 自定义菜单", @"Custom - 自定义弹窗", @"CustomSheet - 类似Sheet效果", @"CustomSheet - 类似Sheet效果2"];
    
    [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:@"cellId"];
    self.tableView.estimatedRowHeight = 44.0;
    
    // customSheetView
    FWCustomSheetViewProperty *property = [[FWCustomSheetViewProperty alloc] init];
    property.popupViewItemHeight = 40;
    property.selectedIndex = 1;
    
    NSArray *itemTitles = @[@"EOS", @"DICE", @"ZKS"];
    
    self.customSheetView = [FWCustomSheetView sheetWithHeaderTitle:@"选择代币" itemTitles:itemTitles itemSecondaryTitles:nil itemImages:nil itemBlock:^(FWPopupView * popupView, NSInteger index, NSString *title) {
        
        NSLog(@"当前选中了：%@", title);
        
    } property:property];
    
    
    // customSheetView
    FWCustomSheetViewProperty *property2 = [[FWCustomSheetViewProperty alloc] init];
    property2.lastNeedAccessoryView = YES;

    NSArray *itemTitles2 = @[@"eoslinwu1122", @"eoslinwusd23", @"导入钱包"];
    NSArray *itemSecondaryTitles2 = @[@"EOS6sHTCXbm4Gz5WRhKxuuBgVZYttvM9tEdU6ThH6kseMWLYDTk9q", @"EOS1sdksbm4Gz5WRhKxuuBgVZYttvM9tEdU6ThH6kseMWLYDTk9q", @""];

    self.customSheetView2 = [FWCustomSheetView sheetWithHeaderTitle:@"选择一个钱包" itemTitles:itemTitles2 itemSecondaryTitles:itemSecondaryTitles2 itemImages:nil itemBlock:^(FWPopupView * popupView, NSInteger index, NSString *title) {

        NSLog(@"当前选中了：%@", title);

    } property:property2];
}

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.titleArray.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cellId"];
    cell.textLabel.text = self.titleArray[indexPath.row];
    cell.textLabel.numberOfLines = 0;
    if (indexPath.row == 9 || indexPath.row == 10) {
        cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
    } else {
        cell.accessoryType = UITableViewCellAccessoryNone;
    }
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    switch (indexPath.row) {
        case 0:
        {
            FWAlertView *alertView = [FWAlertView alertWithTitle:@"标题" detail:@"描述描述描述描述" confirmBlock:^(FWPopupView *popupView, NSInteger index, NSString *title) {
                
            }];
            [alertView show];
        }
            break;
        case 1:
        {
            FWAlertView *alertView = [FWAlertView alertWithTitle:@"标题" detail:@"描述描述描述描述描述描述描述描述描述描述" confirmBlock:^(FWPopupView *popupView, NSInteger index, NSString *title) {
                NSLog(@"点击了确定");
            } cancelBlock:^(FWPopupView *popupView, NSInteger index, NSString *title) {
                NSLog(@"点击了确定");
            }];
            
            [alertView showWithPopupStateBlock:^(FWPopupView * popupView, FWPopupViewState popupViewState) {
                
            }];
        }
            break;
        case 2:
        {
            id block = ^(FWPopupView *popupView, NSInteger index, NSString *title){
                NSLog(@"AlertView：点击了第 %ld 个按钮", (long)index);
            };
            
            // 注意：此时“确定”按钮是不让按钮自己隐藏的
            NSArray *items = @[[[FWPopupItem alloc] initWithTitle:@"取消" itemType:FWItemTypeNormal isCancel:YES canAutoHide:YES itemTitleColor:kRGB(141, 151, 163) itemBackgroundColor:nil itemClickedBlock:block],
                               [[FWPopupItem alloc] initWithTitle:@"确定" itemType:FWItemTypeNormal isCancel:NO canAutoHide:YES itemTitleColor:kRGB(29, 150, 227) itemBackgroundColor:nil itemClickedBlock:block]];
            
            FWAlertViewProperty *vProperty = [[FWAlertViewProperty alloc] init];
            vProperty.alertViewWidth = MAX([UIScreen mainScreen].bounds.size.width * 0.65, 275);
            vProperty.titleFontSize = 17.0;
            vProperty.detailFontSize = 14.0;
            vProperty.detailColor = kRGB(141, 151, 163);
            vProperty.buttonFontSize = 14.0;
            // 还有很多参数可设置...
            
            FWAlertView *alertView = [FWAlertView alertWithTitle:@"标题" detail:@"描述描述描述描述描述描述描述描述描述描述" inputPlaceholder:nil keyboardType:UIKeyboardTypeDefault isSecureTextEntry:NO customView:nil items:items vProperty:vProperty];
            [alertView show];
        }
            break;
        case 3:
        {
            id block = ^(FWPopupView *popupView, NSInteger index, NSString *title){
                NSLog(@"AlertView：点击了第 %ld 个按钮", (long)index);
            };
            
            NSArray *items = @[[[FWPopupItem alloc] initWithTitle:@"取消" itemType:FWItemTypeNormal isCancel:YES canAutoHide:YES itemClickedBlock:block],
                               [[FWPopupItem alloc] initWithTitle:@"确定" itemType:FWItemTypeNormal isCancel:NO canAutoHide:YES itemClickedBlock:block],
                               [[FWPopupItem alloc] initWithTitle:@"其他" itemType:FWItemTypeNormal isCancel:NO canAutoHide:YES itemClickedBlock:block]];
            
            FWAlertView *alertView = [FWAlertView alertWithTitle:@"标题" detail:@"描述描述描述描述描述描述描述描述描述描述" inputPlaceholder:nil keyboardType:UIKeyboardTypeDefault isSecureTextEntry:NO customView:nil items:items];
            [alertView show];
        }
            break;
        case 4:
        {
            id block = ^(FWPopupView *popupView, NSInteger index, NSString *title){
                NSLog(@"AlertView：点击了第 %ld 个按钮", (long)index);
            };
            NSArray *items = @[[[FWPopupItem alloc] initWithTitle:@"取消" itemType:FWItemTypeNormal isCancel:YES canAutoHide:YES itemClickedBlock:block],
                               [[FWPopupItem alloc] initWithTitle:@"确定" itemType:FWItemTypeNormal isCancel:NO canAutoHide:YES itemClickedBlock:block]];
            
            FWAlertView *alertView = [FWAlertView alertWithTitle:@"标题" detail:@"带输入框" inputPlaceholder:@"请输入..." keyboardType:UIKeyboardTypeDefault isSecureTextEntry:YES customView:nil items:items];
            [alertView show];
        }
            break;
        case 5:
        {
            __weak typeof(self) weakSelf = self;
            id block = ^(FWPopupView *popupView, NSInteger index, NSString *title){
                if (index == 1)
                {
                    // 这边演示了如何手动去调用隐藏
                    [weakSelf.alertWithImageView hide];
                }
            };
            
            // 注意：此时“确定”按钮是不让按钮自己隐藏的
            NSArray *items = @[[[FWPopupItem alloc] initWithTitle:@"取消" itemType:FWItemTypeNormal isCancel:YES canAutoHide:YES itemClickedBlock:block],
                               [[FWPopupItem alloc] initWithTitle:@"确定" itemType:FWItemTypeNormal isCancel:NO canAutoHide:NO itemClickedBlock:block]];
            UIImageView *customImageView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"audio_bgm_4"]];
            
            self.alertWithImageView = [FWAlertView alertWithTitle:@"标题" detail:@"带输入框" inputPlaceholder:nil keyboardType:UIKeyboardTypeDefault isSecureTextEntry:NO customView:customImageView items:items];
            [self.alertWithImageView show];
        }
            break;
        case 6:
        {
            NSArray *items = @[@"Sheet0", @"Sheet1", @"Sheet2", @"Sheet3"];
            
            FWSheetView *sheetView = [FWSheetView sheetWithTitle:nil itemTitles:items itemBlock:^(FWPopupView *popupView, NSInteger index, NSString *title) {
                NSLog(@"Sheet：点击了第 %ld 个按钮", (long)index);
            } cancenlBlock:^{
                NSLog(@"点击了取消");
            }];
            sheetView.vProperty.touchWildToHide = @"1";
            [sheetView show];
        }
            break;
        case 7:
        {
            NSArray *items = @[@"Sheet0", @"Sheet1", @"Sheet2", @"Sheet4", @"Sheet5", @"Sheet6", @"Sheet7", @"Sheet8", @"Sheet9", @"Sheet10", @"Sheet11", @"Sheet12", @"Sheet13", @"Sheet14"];
            
            FWSheetView *sheetView = [FWSheetView sheetWithTitle:@"标题" itemTitles:items itemBlock:^(FWPopupView *popupView, NSInteger index, NSString *title) {
                NSLog(@"Sheet：点击了第 %ld 个按钮", (long)index);
            } cancenlBlock:^{
                NSLog(@"点击了取消");
            }];
            [sheetView show];
        }
            break;
        case 8:
        {
            FWDateView *dateView = [FWDateView dateWithConfirmBlock:^(UIDatePicker *datePicker) {
                NSLog(@"点击了 FWDateView 的确定");
            } cancelBlock:^{
                NSLog(@"点击了 FWDateView 的取消");
            }];
            [dateView show];
        }
            break;
        case 9:
        {
            [self.navigationController pushViewController:[[FWMenuViewDemoVC alloc] init] animated:YES];
        }
            break;
        case 10:
        {
            [self.navigationController pushViewController:[[FWCustomDemoVC alloc] init] animated:YES];
        }
            break;
        case 11:
        {
            [self.customSheetView show];
        }
            break;
        case 12:
        {
            [self.customSheetView2 show];
        }
            break;
            
        default:
            break;
    }
}

@end

File: FWPopupView/FWPopupView/FWDemoViewController.swift
//
//  FWDemoViewController.swift
//  FWPopupView
//
//  Created by xfg on 2018/3/26.
//  Copyright © 2018年 xfg. All rights reserved.
//

import Foundation
import UIKit

class FWDemoViewController: UITableViewController {
    
    var alertImage: FWAlertView!
    
    /// 注意：这边不同的示例可能还附加演示了一些特性（比如：遮罩层是否能够点击、遮罩层的背景颜色等等），有用到时可以参考
    var titleArray = ["Alert - 单个按钮", "Alert - 两个按钮", "Alert - 两个按钮（修改参数）", "Alert - 多个按钮", "Alert - 带输入框", "Alert - 带自定义视图", "Sheet - 少量Item", "Sheet - 大量Item", "Date - 自定义日期选择", "Menu - 自定义菜单", "Custom - 自定义弹窗", "CustomSheet - 类似Sheet效果", "CustomSheet - 类似Sheet效果2", "RadioButton"]
    
    let block: FWPopupItemClickedBlock = { (popupView, index, title) in
        print("AlertView：点击了第\(index)个按钮")
    }
    
    lazy var customSheetView: FWCustomSheetView = {
        
        let property = FWCustomSheetViewProperty()
        property.popupViewItemHeight = 40
        
        let titles = ["EOS", "DICE", "ZKS"]
        
        let customSheetView = FWCustomSheetView.sheet(headerTitle: "选择代币", itemTitles: titles, itemSecondaryTitles: nil, itemImages: nil, itemBlock: { (popupView, index, title) in
            print("customSheet：点击了第\(index)个按钮")
        }, property: property)
        
        return customSheetView
    }()
    
    lazy var customSheetView2: FWCustomSheetView = {
        
        let property = FWCustomSheetViewProperty()
        property.lastNeedAccessoryView = true
        
        let titles = ["fdksds123123", "fdksds112233", "导入钱包"]
        let secondaryTitles = ["EOS6sHTCXbm4Gz5WRhKxuuBgVZYttvM9tEdU6ThH6kseMWLYDTk9q", "EOS1sdksbm4Gz5WRhKxuuBgVZYttvM9tEdU6ThH6kseMWLYDTk9q",""]
        let images = [UIImage(named: "right_menu_addFri"),
                      UIImage(named: "right_menu_addFri"),
                      UIImage(named: "right_menu_multichat")]
        
        let customSheetView = FWCustomSheetView.sheet(headerTitle: "选择一个钱包", itemTitles: titles, itemSecondaryTitles: secondaryTitles, itemImages: images as? [UIImage], itemBlock: { (popupView, index, title) in
            print("customSheet：点击了第\(index)个按钮")
        }, property: property)
        
        return customSheetView
    }()
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.navigationItem.title = "FWPopupView"
        
        self.tableView.estimatedRowHeight = 44.0
        self.tableView.rowHeight = UITableView.automaticDimension
        self.tableView.register(UITableViewCell.self, forCellReuseIdentifier: "cellId")
        
        // 设置弹窗外部可点击
        // FWPopupWindow.sharedInstance.touchWildToHide = true
    }
}

extension FWDemoViewController {
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        
        return titleArray.count
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        let cell = UITableViewCell(style: .default, reuseIdentifier: "cellId")
        cell.textLabel?.text = titleArray[indexPath.row]
        cell.textLabel?.numberOfLines = 0
        if indexPath.row == 9 || indexPath.row == 10 {
            cell.accessoryType = .disclosureIndicator
        } else if indexPath.row == 13 {
            let property = FWRadioButtonProperty()
            property.selectedStateColor = UIColor.red
            property.buttonType = .rectangle
            cell.accessoryView = FWRadioButton.radio(frame: CGRect(x: 0, y: 0, width: 25, height: 25), property: property)
        } else {
            cell.accessoryType = .none
            cell.accessoryView = nil
        }
        return cell
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        
        switch indexPath.row {
        case 0:
            let alertView = FWAlertView.alert(title: "标题", detail: "描述描述描述描述") { (popupView, index, title) in
                print("点击了确定")
            }
            alertView.show()
            break
        case 1:
            let alertView = FWAlertView.alert(title: "标题", detail: "描述描述描述描述描述描述描述描述描述描述", confirmBlock: { (popupView, index, title) in
                print("点击了确定")
            }, cancelBlock: { (popupView, index, title) in
                print("点击了取消")
            })
            // 设置AlertView外部背景色
            alertView.attachedView?.fwMaskViewColor = UIColor(white: 0, alpha: 0.2)
            alertView.show { (popupView, popupViewState) in
                print("当前弹窗状态：\(popupViewState.rawValue)")
                if popupViewState == .didDisappear {
                    // 隐藏时把背景色改回来（这个非必要，我这边只是一个演示）
                    alertView.attachedView?.fwMaskViewColor = UIColor(white: 0, alpha: 0.5)
                }
            }
            break
        case 2:
            // 注意：此时“确定”按钮是不让按钮自己隐藏的
            let items = [FWPopupItem(title: "取消", itemType: .normal, isCancel: true, canAutoHide: true, itemTitleColor: kPV_RGBA(r: 141, g: 151, b: 163, a: 1.0), itemBackgroundColor: nil, itemClickedBlock: block),
                         FWPopupItem(title: "确定", itemType: .normal, isCancel: false, canAutoHide: true, itemTitleColor: kPV_RGBA(r: 29, g: 150, b: 227, a: 1.0), itemBackgroundColor: nil, itemClickedBlock: block)]
            
            // 主要演示修改参数
            let vProperty = FWAlertViewProperty()
            vProperty.alertViewWidth = max(UIScreen.main.bounds.width * 0.65, 275)
            vProperty.titleFontSize = 17.0
            vProperty.detailFontSize = 14.0
            vProperty.detailColor = kPV_RGBA(r: 141, g: 151, b: 163, a: 1.0)
            vProperty.buttonFontSize = 14.0
            vProperty.maskViewColor = UIColor(white: 0, alpha: 0.5)
            vProperty.touchWildToHide = "1"
            // 还有很多参数可设置...
            
            let alertView = FWAlertView.alert(title: "标题", detail: "描述描述描述描述描述描述描述描述描述描述", inputPlaceholder: nil, keyboardType: .default, isSecureTextEntry: false, customView: nil, items: items, vProperty: vProperty)
            
            alertView.show()
            break
        case 3:
            let items = [FWPopupItem(title: "取消", itemType: .normal, isCancel: true, canAutoHide: true, itemClickedBlock: block),
                         FWPopupItem(title: "确定", itemType: .normal, isCancel: false, canAutoHide: true, itemClickedBlock: block),
                         FWPopupItem(title: "其他", itemType: .normal, isCancel: false, canAutoHide: true, itemClickedBlock: block)]
            
            let alertView = FWAlertView.alert(title: "标题", detail: "描述描述描述描述描述描述描述描述描述描述", inputPlaceholder: nil, keyboardType: .default, isSecureTextEntry: false, customView: nil, items: items)
            alertView.show()
            break
        case 4:
            let items = [FWPopupItem(title: "取消", itemType: .normal, isCancel: true, canAutoHide: true, itemClickedBlock: block),
                         FWPopupItem(title: "确定", itemType: .normal, isCancel: false, canAutoHide: true, itemClickedBlock: block)]
            
            let alertView = FWAlertView.alert(title: "标题", detail: "带输入框", inputPlaceholder: "请输入...", keyboardType: .numberPad, isSecureTextEntry: true, customView: nil, items: items)
            alertView.show()
            break
        case 5:
            if self.alertImage == nil {
                
                let block2: FWPopupItemClickedBlock = { [weak self] (popupView, index, title) in
                    
                    if index == 1 {
                        // 这边演示了如何手动去调用隐藏
                        self?.alertImage.hide()
                    }
                }
                
                // 注意：此时“确定”按钮是不让按钮自己隐藏的
                let items = [FWPopupItem(title: "取消", itemType: .normal, isCancel: true, canAutoHide: true, itemClickedBlock: block2),
                             FWPopupItem(title: "确定", itemType: .normal, isCancel: false, canAutoHide: false, itemClickedBlock: block2)]
                // 注意：添加自定义的视图，需要设置确定的Frame值
                let customImageView = UIImageView(image: UIImage(named: "audio_bgm_4"))
                
                let vProperty = FWAlertViewProperty()
                vProperty.touchWildToHide = "1"
                
                self.alertImage = FWAlertView.alert(title: "标题", detail: "带自定义视图", inputPlaceholder: nil, keyboardType: .default, isSecureTextEntry: false, customView: customImageView, items: items, vProperty: vProperty)
            }
            
            self.alertImage.show()
            break
        case 6:
            let items = ["Sheet0", "Sheet1", "Sheet2", "Sheet3"]
            
            let vProperty = FWSheetViewProperty()
            vProperty.touchWildToHide = "1"
            
            let sheetView = FWSheetView.sheet(title: nil, itemTitles: items, itemBlock: { (popupView, index, title) in
                print("Sheet：点击了第\(index)个按钮")
            }, cancenlBlock: {
                print("点击了取消")
            }, property: vProperty)
            sheetView.show()
            break
        case 7:
            let items = ["Sheet0", "Sheet1", "Sheet2", "Sheet3", "Sheet4", "Sheet5", "Sheet6", "Sheet7", "Sheet8", "Sheet9", "Sheet10", "Sheet11", "Sheet12", "Sheet13", "Sheet14"]
            
            let sheetView = FWSheetView.sheet(title: "标题", itemTitles: items, itemBlock: { (popupView, index, title) in
                print("Sheet：点击了第\(index)个按钮，名称为：\(String(describing: title))")
            }, cancenlBlock: {
                print("点击了取消")
            })
            sheetView.show()
            break
        case 8:
            let dateView = FWDateView.date(confirmBlock: { (datePicker) in
                print("当前选定时间：\(datePicker.date)")
            }, cancelBlock: {
                print("点击了 FWDateView 的取消")
            })
            dateView.datePicker.minimumDate = Date()
            dateView.datePicker.locale = Locale(identifier: "zh_Hans_CN")
            dateView.datePicker.datePickerMode = .dateAndTime
            dateView.datePicker.calendar = Calendar.current
            dateView.show()
            break
        case 9:
            self.navigationController?.pushViewController(FWMenuViewDemoVC(), animated: true)
            break
        case 10:
            self.navigationController?.pushViewController(FWCustomPopupDemoVC(), animated: true)
            break
        case 11:
            self.customSheetView.show()
            break
        case 12:
            self.customSheetView2.show()
            break
        default:
            break
        }
    }
}
